libs/constrained-generators/src/Constrained/Base.hs

```diff
@@ -3683,13 +3780,26 @@ instance (Ord a, HasSpec fn a) => Semigroup (SetSpec fn a) where
 instance (Ord a, HasSpec fn a) => Monoid (SetSpec fn a) where
   mempty = SetSpec mempty mempty TrueSpec
 
-guardSetSpec :: (HasSpec fn a, Ord a) => SetSpec fn a -> Specification fn (Set a)
-guardSetSpec (SetSpec must elem ((<> geqSpec 0) -> size))
+guardSetSpec :: (HasSpec fn a, Ord a) => [String] -> SetSpec fn a -> Specification fn (Set a)
```

Comment:
It would be better if we put the `[String]` wrapper in the call site with a call to ExplainSpec

Reply:
The problem with this is that the [String] in the explain will be deep in the message stack, so if an error occurs, it will appear far away from the problem that caused the error. By passing the [String] as an argument to guardTypeSpec method instances, it can decide where best to incorporate the [String],  This represents the context of the call, and most guardTypeSpec methods are called in more than one place so the context can be different. Of you look at the messages for some deliberately failing examples, It really helps for the [String] to be passed in.